### ISSUE TITLE
RFC: remove block API

### ISSUE THREAD

**Issue by alanshaw** [ipfs-shipyard/ipfsx#8]
Everything we can do with the block API we can do with the DAG API, and the DAG API prevents us from putting invalid data in IPFS.

**Comment by daviddias** [comment]
Not quite. Blocks are raw binary blobs and it is what Bitswap (the Block Exchange) understands to trade data. It just so happens that as a user, you pretty much don't have any reason to use the Block API unless you don't want to use IPLD at all.

**Comment by alanshaw** [comment]
👍 understood - could it be a private API that JS IPFS uses internally or is there something that necessitates it being public?

**Comment by daviddias** [comment]
I would guess that most of the users won't need it. If we learn that having that available is hurting UX/DX of the APIs, that it can make sense to hide/disable it.
